File: e2e/playwright-typescript/interactions/pages/CreateTaskForm.ts
import { Locator, Page } from '@playwright/test';
import { TaskPriority } from '@/data/TaskPriority';
import { TaskStatus } from '@/data/TaskStatus';

export class CreateTaskForm {
  readonly titleInput: Locator;
  readonly descriptionInput: Locator;
  readonly statusDropdown: Locator;
  readonly priorityDropdown: Locator;
  readonly submitButton: Locator;

  constructor(private readonly page: Page) {
    this.titleInput = page.getByTestId('create-task-title-input');
    this.descriptionInput = page.locator('#description');
    this.statusDropdown = page.getByTestId('status-dropdown');
    this.priorityDropdown = page.getByTestId('priority-dropdown');
    this.submitButton = page.getByTestId('create-button');
  }

  statusOption(status: TaskStatus): Locator {
    return this.page.getByTestId(`status-dropdown-option-${status}`);
  }

  priorityOption(priority: TaskPriority): Locator {
    return this.page.getByTestId(`priority-dropdown-option-${priority}`);
  }
}

File: demo-android/app/src/androidTest/java/com/example/demo/e2e/interaction/page/CreateTaskForm.kt
package com.example.demo.e2e.interaction.page

import androidx.compose.ui.test.junit4.AndroidComposeTestRule
import com.example.demo.MainActivity
import com.example.demo.data.model.TaskPriority
import com.example.demo.data.model.TaskStatus

class CreateTaskForm(
    rule: AndroidComposeTestRule<*, MainActivity>,
) : ComposePage(rule) {
    fun createButton() = node("create-button")

    fun titleField() = node("create-task-title-input")

    fun descriptionField() = node("task-description-input")

    fun statusDropdown() = node("status-dropdown")

    fun priorityDropdown() = node("priority-dropdown")

    fun statusOption(status: TaskStatus) = node("status-dropdown-option-${status.name}")

    fun priorityOption(priority: TaskPriority) = node("priority-dropdown-option-${priority.name}")
}
